src/Corax/Indexing/IndexedField.cs

```diff
@@ -78,6 +109,13 @@ public override string ToString()
             EntryToTerms = new();
     }
 
+    public IndexedField CreateVirtualIndexedField(IndexFieldBinding dynamicField)
+    {
+        return new IndexedField(dynamicField.FieldId, dynamicField.FieldName, dynamicField.FieldNameLong, dynamicField.FieldNameDouble,
+            dynamicField.FieldTermTotalSumField, dynamicField.Analyzer, dynamicField.FieldIndexingMode, dynamicField.HasSuggestions, dynamicField.ShouldStore,
+            SupportedFeatures, dynamicField.FieldNameForStatistics, FieldRootPage, TermsVectorFieldRootPage, Storage, Textual, Longs, Doubles, this);
+    }
+
```

Comment:
We had underlying issue when `IndexedField` from known field and dynamic shared the same name. In such scenario, we previously skipped overwritten dynamic field configuration and returned the `IndexedField` from known mapping. We cannot just create duplicate `IndexedField` in such scenario since we're modifying the trees twice in same tx and frequency wouldn't match. I've developed the idea of a `virtual` `IndexedField` for this scenario. Now, all underlying data structures containing terms are now shared between known and dynamic field with the same name. In such scenario we're skipping virtual during commit since all terms from dynamic will be written when committing the known field. This approach is useful for `IndexEntryBuilder`, allowing us to apply custom analyzer/storage configuration, though at the end they're both referencing to the exactly same CompactTree/FST anyway.

cc @redknightlois @ayende

Reply:
I like it.